src/ranged.cpp

```diff
@@ -1456,6 +1463,12 @@ double player::get_weapon_dispersion(item *weapon, bool random) const
 int recoil_add(player &p, const item &gun)
 {
     int ret = gun.gun_recoil();
+    if ( weapon->has_flag("MOUNTED_GUN") &&
+        !m.has_flag_ter_or_furn( "MOUNTABLE", u.pos3() ) &&
```

Comment:
Same comment for `m` and `veh` as above.

Use `p.pos()` instead of `u.pos3()`. (This function may be called on a NPC object where `p` is the NPC, in that case it must use the position of the NPC.)